src/services/identity/__tests__/SitesService.spec.ts

```diff
@@ -149,26 +610,176 @@ describe("SitesService", () => {
     })
   })
 
+  describe("getMergeAuthorEmail", () => {
+    it("should return the email of the merge commit author if it was not performed using the common access token", async () => {
+      // Arrange
+      const expected = MOCK_GITHUB_EMAIL_ADDRESS_ONE
+      const commit: GitHubCommitData = {
+        author: {
+          name: MOCK_GITHUB_NAME_ONE,
+          email: MOCK_GITHUB_EMAIL_ADDRESS_ONE,
+          date: MOCK_GITHUB_DATE_ONE,
+        },
+        message: MOCK_COMMIT_MESSAGE_ONE,
+      }
+
+      // Act
+      const actual = await SitesService.getMergeAuthorEmail(
+        commit,
+        mockUserWithSiteSessionData
+      )
+
+      // Assert
+      expect(actual).toEqual(expected)
```

Comment:
should we assert that `extractAuthorEmail` was called with the commit? (similarly for below tests)

Reply:
Wait is that possible? My understanding was that the `expect().toHaveBeenCalled()` only works for a mocked function, but `extractAuthorEmail` is an actual function that is undergoing testing as part of this test file.

Reply:
yeap, what we need to do here is to construct a [spy](https://jestjs.io/docs/jest-object#jestspyonobject-methodname). this would then allow us to call assertions on the spy transparently.

Reply:
yeap, what we need to do is to construct a [spy](https://jestjs.io/docs/jest-object#jestspyonobject-methodname), which we can then use to make assertions on transparently

Reply:
TIL, thanks for sharing! Added in c333d8c.